node/src/blockchain/blockchain_interface.rs

```diff
@@ -2339,30 +2321,34 @@ mod tests {
             ),
             BlockchainError::UninitializedBlockchainInterface,
         ];
-        let pretty_print_closure = |err_to_resolve: &BlockchainError| match err_to_resolve {
-            BlockchainError::InvalidUrl => (err_to_resolve.to_string(), 11),
-            BlockchainError::InvalidAddress => (err_to_resolve.to_string(), 22),
-            BlockchainError::InvalidResponse => (err_to_resolve.to_string(), 33),
-            BlockchainError::QueryFailed(..) => (err_to_resolve.to_string(), 44),
-            BlockchainError::UninitializedBlockchainInterface => (err_to_resolve.to_string(), 55),
-        };
 
-        let actual_error_msgs = collect_match_displayable_error_variants_in_exhaustive_mode(
-            original_errors.as_slice(),
-            pretty_print_closure,
-            vec![11, 22, 33, 44, 55],
-        );
+        let actual_error_msgs = original_errors
+            .iter()
+            .map(|err| err.to_string())
+            .collect::<Vec<_>>();
 
+        let original_errors_len = original_errors.len();
+        assert_eq!(
+            original_errors_len,
+            BlockchainError::VARIANT_COUNT,
+            "you forgot to add all variants in this test"
+        );
+        let check_hash_set: HashSet<BlockchainError> = HashSet::from_iter(original_errors);
```

Comment:
I think this check is unnecessary, and I can't think of any way to force the future developer to do what we want here.